internal/provider/waypoint/resource_waypoint_add_on_definition.go

```diff
@@ -0,0 +1,563 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package waypoint
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-waypoint-service/preview/2023-08-18/client/waypoint_service"
+	waypointmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-waypoint-service/preview/2023-08-18/models"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/listplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &AddOnDefinitionResource{}
+var _ resource.ResourceWithImportState = &AddOnDefinitionResource{}
+
+func NewAddOnDefinitionResource() resource.Resource {
+	return &AddOnDefinitionResource{}
+}
+
+// AddOnDefinitionResource defines the resource implementation.
+type AddOnDefinitionResource struct {
+	client *clients.Client
+}
+
+// AddOnDefinitionResourceModel describes the resource data model.
+type AddOnDefinitionResourceModel struct {
+	ID                     types.String `tfsdk:"id"`
+	Name                   types.String `tfsdk:"name"`
+	ProjectID              types.String `tfsdk:"project_id"`
+	OrgID                  types.String `tfsdk:"organization_id"`
+	Summary                types.String `tfsdk:"summary"`
+	Labels                 types.List   `tfsdk:"labels"`
+	Description            types.String `tfsdk:"description"`
+	ReadmeMarkdownTemplate types.String `tfsdk:"readme_markdown_template"`
+
+	TerraformCloudWorkspace *tfcWorkspace    `tfsdk:"terraform_cloud_workspace_details"`
+	TerraformNoCodeModule   *tfcNoCodeModule `tfsdk:"terraform_no_code_module"`
+
+	// questionable
+	// Namespace types.String `tfsdk:"namespace_id"`
```

Comment:
Dead code. I'm sure you copied this from mine 😆 I'll remove it in my code as well